hare/broker.go

```diff
@@ -62,15 +58,11 @@ func newBroker(
 		pubKeyExtractor: pubKeyExtractor,
 		roleValidator:   roleValidator,
 		stateQuerier:    stateQuerier,
-		nodeSyncState:   syncState,
 		mchOut:          mch,
 		outbox:          make(map[uint32]chan any),
-		pending:         make(map[uint32][]any),
 		latestLayer:     types.GetEffectiveGenesis(),
 		limit:           limit,
-		minDeleted:      types.GetEffectiveGenesis(),
 	}
-	b.ctx, b.cancel = context.WithCancel(context.Background())
```

Comment:
why remove this?

it is meant for it to be safe to call Close() without calling Start()

Reply:
My thinking was that it's a mistake to:
* create a broker and then close it without calling Start since it won't have done any work
* create a broker and handle messages before calling start
So it's better to disallow those flows. Now we'll get a panic if we try to close the broker or handle messages before calling Start, which will allow us to address the problem

Reply:
> So it's better to disallow those flows. Now we'll get a panic if we try to close the broker or handle messages before calling Start, which will allow us to address the problem

i don't see this change as "better". the problems you described are quite unimportant and would be a bug in the startup flow. the fact that `Close()` is called means the app is shutting down. having a panic during the shutdown flow is .... distracting from the actual cause that trigger it to shutdown in the first place.

i think a better change would be initializing broker with a real context obj. 

in any case, i am also fine with this change.